### PR TITLE
Feature/jra 90 i os download and store all the information from the current city

### PR DESCRIPTION
**Pull request by YawarValeriano**
Added manager for getting all data from firebase
Added core data models and computed properties
Modified timer to finish after 3 seconds or after storing data into Core Data
Added unit test

### CODE REVIEW

File: RoutesApp_ios/modules/citypicker/controllers/CitySplashViewController.swift [hunk 1]
```diff
@@ -11,12 +11,15 @@ class CitySplashViewController: UIViewController {
 
     @IBOutlet weak var cityNameLabel: UILabel!
     var timer = Timer()
+    var timer2 = Timer()
```
Comment (jmacboy) [inline RoutesApp_ios/modules/citypicker/controllers/CitySplashViewController.swift]: is this variable still used?

File: RoutesApp_ios/repository/local/managers/LocalDataManager.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    completion(.failure(error))
+                }
+            }
+        } else {
+            completion(.failure(CoreDataError.Failed(cause: String.localizeString(localizedString: ConstantVariables.errorUpdatingData))))
+        }
+
+    }
+
+    private func getAndSaveTourpointCategoriesWithTourpoints(tourpoints: [Tourpoint], completion: @escaping(Result<Void, Error>) -> Void) {
```
Comment (jmacboy) [inline RoutesApp_ios/repository/local/managers/LocalDataManager.swift:40]: doesn't this already has a manager for tour points where it should be?

File: RoutesApp_ios/repository/local/managers/LocalDataManager.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                try? self.context.save()
+                self.getLineCategories(completion: completion)
+
+            case.failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    private func getLineCategories(completion: @escaping(Result<Void, Error>) -> Void) {
```
Comment (jmacboy) [inline RoutesApp_ios/repository/local/managers/LocalDataManager.swift:62]: same question but with lineCategories

File: RoutesApp_ios/repository/local/managers/LocalDataManager.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                    category.toEntity(context: self.coreDataManager.getContext())
+                }
+                self.getLines(categories: categories, completion: completion)
+            case.failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    private func getLines(categories: [LinesCategory], completion: @escaping(Result<Void, Error>) -> Void) {
```
Comment (jmacboy) [inline RoutesApp_ios/repository/local/managers/LocalDataManager.swift:76]: same question but with lines

File: RoutesApp_ios/repository/local/managers/LocalDataManager.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                                                   context: self.coreDataManager.getContext())
+                    self.getLineRoutes(line: lineEntity, completion: completion)
+                }
+            case.failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    private func getLineRoutes(line: LineEntity, completion: @escaping(Result<Void, Error>) -> Void) {
```
Comment (jmacboy) [inline RoutesApp_ios/repository/local/managers/LocalDataManager.swift:92]: same question but with lineRoutes

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        }
+    }
+
+    override func reverseTransformedValue(_ value: Any?) -> Any? {
+        guard let data = value as? Data else { return nil }
+        do {
+            let coordinate = try NSKeyedUnarchiver.unarchivedObject(ofClass: NSCoordinates.self, from: data)
+            return coordinate
+        } catch {
+            print(error)
```
Comment (IrvingJVR-MX) [inline RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift]: i think you should remove the print

File: RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+class CoordinateTransformer: ValueTransformer {
+
+    override func transformedValue(_ value: Any?) -> Any? {
+        guard let coordinate = value as? NSCoordinates else { return nil }
+        do {
+            let data = try NSKeyedArchiver.archivedData(withRootObject: coordinate, requiringSecureCoding: true)
+            return data
+        } catch {
+            print(error)
```
Comment (IrvingJVR-MX) [inline RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift]: i think you should remove the print

File: RoutesApp_ios/repository/local/managers/LocalDataManager.swift (hunk 2 above)
Comment (YawarValeriano) [reply RoutesApp_ios/repository/local/managers/LocalDataManager.swift:40]: I'm using the corresponding managers to get the data but the way I handle the result is different, that's why I have this logic in this manager

File: RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift (hunk 6 above)
Comment (YawarValeriano) [reply RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift]: removed

File: RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift (hunk 7 above)
Comment (YawarValeriano) [reply RoutesApp_ios/utilities/helpers/CoordinateTransformer.swift]: removed

File: RoutesApp_ios/modules/citypicker/controllers/CitySplashViewController.swift (hunk 1 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/citypicker/controllers/CitySplashViewController.swift]: Removed

**Review by IrvingJVR-MX — APPROVED** [review]

**Review by jmacboy — APPROVED** [review]

**Pull request merged**